### PR TITLE
Add student sidebar directive components.

### PR DESCRIPTION
**Pull request by kennyczadzeck**
Sidebar directive for students to display all modules and lessons within their library.

### CODE REVIEW

**Comment by krish-andres** [comment]
Looks good!

File: assets/js/directives/student_sidebar_directive.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+app.directive("studentSidebarDirective", ["$route", "Module", function($route, Module) {
+  return {
+    templateUrl: "/js/templates/student_sidebar_template.html",
+    link: function(scope, elem, attr) {
+      scope.params = $route.current.params;
```
Comment (jplouie) [inline assets/js/directives/student_sidebar_directive.js]: you don't need to put variables on the scope object if you're not displaying them. also, you're not using this variable

File: assets/js/templates/student_sidebar_template.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      </div>
+      <div id="collapse{{module.id}}" class="panel-collapse collapse">
+          <div class="panel-body">
+              <ul>
+                <li ng-repeat="lesson in module.lessons">
+                  <a href="/#/library/{{module.library.id}}/module/{{module.id}}/lesson/{{lesson.id}}">{{lesson.title}}</a>
+                </li>
+              </ul>
+          </div>
+      </div>
```
Comment (jplouie) [inline assets/js/templates/student_sidebar_template.html]: check your spacing.. it looks like some are 2 spaced tabs and some are 4 spaced tabs

**Comment by kennyczadzeck** [comment]
Made changes based on Jeff's comments.

**Comment by delasky** [comment]
squash and we will merge

**Pull request merged**